Add rules on moving content to collections outside of Ansible (#1971)
ansible/ansible-documentation: docs/docsite/rst/community/collection_contributors/collection_requirements.rst

See :ref:`Migrating content to a different collection <migrate_to_collection>` for complete details.

> Generally we do not object to moving content between collections, or moving content from collections included in Ansible to collections outside the Ansible package, as long as semantic versioning is not violated. More precisely, replacing content by redirects is only a minor change if the destination collection is the dependency of the source collection of the move. (See :ref:`coll_dependencies` for adding new dependencies to collections included in Ansible.)

Reviewer: ```suggestion
Generally, we do not object to moving content between collections or moving content from collections included in Ansible to collections outside the Ansible package, as long as semantic versioning is not violated. More precisely, replacing content with redirects is only a minor change if the destination collection is a dependency of the original collection and both collections are included in Ansible. (See :ref:`coll_dependencies` for more information about adding new dependencies to collections included in Ansible.)
```

Reviewer: Should we also explicitly state "Content must not be removed from collections in minor or patch releases"? We recently had that problem with ansible.posix.

Author: Generally removing content is always a breaking change unless you add the moved to collection as a dependency. In that case, removal is fine for a minor release, but I don't think it's OK for a bugfix release.

Author: I don't like the "and both collections are included in Ansible" part in the suggestion. It is also a minor change if not both of them are in Ansible, but it's not a valid thing to add a non-Ansible dependency - but for completely different reasons.

Author: I'll commit the remainder of your suggestion now, for that I replicate it here:
```suggestion
Generally, we do not object to moving content between collections or moving content from collections included in Ansible to collections outside the Ansible package, as long as semantic versioning is not violated. More precisely, replacing content with redirects is only a minor change if the destination collection is the dependency of the original collection. (See :ref:`coll_dependencies` for more information about adding new dependencies to collections included in Ansible.)
```

> For the community "catch all" collections, we have slightly different rules. We allow to move content out of community.general and community.network to other collections outside of Ansible under the following conditions:

Reviewer 2: Should we really mention `community.network` here [now that we've decided to deprecate and remove it](https://forum.ansible.com/t/8030/35)?

Author: The collection is still alive until in ~6 months, since until then it will appear in Ansible releases. Also it's still possible that someone revives it. I would keep it in here, it's easy to remove it from here later on.

Reviewer 2: @felixfontein OK for me.

Reviewer: Sounds good to me.

> 1. The new collection is appropriately licensed and does not require a CLA to contribute.

Reviewer: Instead of "does not require a CLA to contribute," can we say "follows the Contributor License Agreements policy" and make that a link to the the "Contributor License Agreements" section? There's more detail there about what constitutes a CLA and also that we allow the DCO.

Author: fe077bba8471c3eae5d8949545fe47ee2682019d

> 2. None of the contributors who contributed to the content in the last 6 months objects in a four-weeks period after the plan to deprecate the module has been announced.

Reviewer: ```suggestion
2. None of the contributors who contributed to the content in the last 6 months object in a four-week period after the plan to deprecate the module has been announced.
```

Should we specify "contributors who contributed nontrivial changes" or something like that, also?

Author: I think it's easier (less discussion on what constitutes a nontrivial change) to simply allow all to veto.

> 3. There is a deprecation period of at least 6 months during which deprecation warnings are shown. The deprecation notice must mention that the content is moved to a collection outside the Ansible community package, and that users need to install that collection separately.

Reviewer: ```suggestion
3. There is a deprecation period of at least 6 months during which deprecation warnings are shown. The deprecation notice must mention that the content is moved to a collection outside the Ansible community package and that users need to install that collection separately.
```

> 4. If community members or contributors bring up good reasons in these 6 months to not do the move, the Steering Committee will discuss these and vote on them before the content is removed.

Reviewer: ```suggestion
4. If community members or contributors bring up good reasons in these 6 months to cancel the migration, the Steering Committee will discuss these and vote on them before the content is removed.
```
